### PR TITLE
[SPIR-V] Introduce support for 'spirv.Decorations' metadata node in SPIR-V Backend

### PR DESCRIPTION
**Pull request by VyacheslavLevytskyy**
This PR is to introduce support for 'spirv.Decorations' metadata node in SPIR-V Backend.
See also https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/main/docs/SPIRVRepresentationInLLVM.rst that describes `spirv.Decorations` as an important part of SPIRV-friendly LLVM IR.

### CODE REVIEW

File: llvm/lib/Target/SPIRV/SPIRVUtils.h [hunk 1]
```diff
@@ -59,6 +59,10 @@ void buildOpDecorate(Register Reg, MachineInstr &I, const SPIRVInstrInfo &TII,
                      const std::vector<uint32_t> &DecArgs,
                      StringRef StrImm = "");
 
+// Add an OpDecorate instruction by "spirv.Decorations" metadata node
```
Comment (iliya-diyachkov) [inline llvm/lib/Target/SPIRV/SPIRVUtils.h:62]: Please add '.' at the end of the comment.

File: llvm/include/llvm/IR/IntrinsicsSPIRV.td [hunk 2]
```diff
@@ -14,6 +14,7 @@ let TargetPrefix = "spv" in {
   def int_spv_assign_type : Intrinsic<[], [llvm_any_ty, llvm_metadata_ty]>;
   def int_spv_assign_ptr_type : Intrinsic<[], [llvm_any_ty, llvm_metadata_ty, llvm_i32_ty], [ImmArg<ArgIndex<2>>]>;
   def int_spv_assign_name : Intrinsic<[], [llvm_any_ty, llvm_vararg_ty]>;
+  def int_spv_assign_decoration : Intrinsic<[], [llvm_any_ty, llvm_metadata_ty]>;
```
Comment (michalpaszkowski) [inline llvm/include/llvm/IR/IntrinsicsSPIRV.td:17]: Please update the [SPIRVUsage](https://llvm.org/docs/SPIRVUsage.html#target-intrinsics) document with relevant information about the new intrinsic.

Comment (VyacheslavLevytskyy) [reply llvm/include/llvm/IR/IntrinsicsSPIRV.td:17]: Done, thank you

Comment (michalpaszkowski) [reply llvm/include/llvm/IR/IntrinsicsSPIRV.td:17]: Thank you!

**Review by michalpaszkowski — APPROVED** [review]

**Pull request merged**